chronos/asyncloop.nim

```diff
@@ -227,14 +231,27 @@ func getAsyncTimestamp*(a: Duration): auto {.inline.} =
     result = cast[int32](res)
     result += min(1, cast[int32](mid))
 
+template removeTimers(loop) =
+  var timers = cast[seq[TimerCallback]](loop.timers)
+  timers.keepItIf( not(it.removed) )
```

Comment:
This doesn't preserve the properties of the binary heap. Consider the following heap:

```
     1
   /   \
  15    2
 / \   / \
16 20 3   4
```

If 16 and 20 are removed, the simple filtering will leave 3 and 4 as child nodes of 15, which will violate the heap ordering invariant.